### DISCUSSION TITLE
Searching for shapes in Libraries

### DISCUSSION THREAD

**Discussion by uxkjaer** [jgraph/drawio#3387] [Q&A]
First of all, thanks so much for an awesome tool!
I'm using the tool to create architecture diagrams for SAP Business Technology Platform. The SAP community has created a [yeoman generator](https://blogs.sap.com/2022/11/26/business-technology-platform-solution-diagrams-and-icons-taken-to-the-next-level/) to fetch icons and create custom libraries.
When I try to use these libraries for searching, it only shows me the shape if I type in the full name. Would it be possible to have either an option to search for contains as well or maybe even be able to set metadata data to the shapes like tags or keywords to help the search finding the appropriate shapes. I've added this as a question in hopes it might already exist.

**Comment by danielcampetti** [comment]
I need that too!
